mu-plugins/10up-plugin/includes/classes/ModuleInitialization.php

```diff
@@ -56,9 +58,27 @@ private function __construct() {
 	 * @return array<string>
 	 */
 	protected function get_classes() {
-		$class_finder = new ClassFinder();
-		$class_finder::setAppRoot( TENUP_PLUGIN_PATH );
-		return $class_finder::getClassesInNamespace( 'TenUpPlugin', ClassFinder::RECURSIVE_MODE );
+		// Get all classes from this directory and its subdirectories.
+		$class_finder = Discover::in( __DIR__ );
+		// Only fetch classes.
+		$class_finder->classes();
+		// Only fetch classes in the current namespace.
+		$class_finder->custom(
+			fn( DiscoveredStructure $structure ) => str_starts_with( $structure->namespace, __NAMESPACE__ )
+		);
+
+		// If we are in production or staging, cache the class loader to improve performance.
+		if ( in_array( wp_get_environment_type(), [ 'production', 'staging' ], true ) ) {
```

Comment:
**thought:** i think we should assume production and check local. In theory WP hosts could use anything for this value and would not count all possible env types like preprod, uat, qa, etc that some proojects use

Reply:
@tobeycodes `wp_get_environment_type()` only has 4 available options, using anything other than those will default to production. See: https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-includes/load.php#L229-L234

Reply:
Oh... I should have read the source code before commenting. This makes sense then